modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -675,12 +676,28 @@ public ServiceDetailVo getServiceDetailsByIdForIsv(UUID id) {
         }
         Optional<String> namespace = identityProviderManager.getUserNamespace();
         if (namespace.isEmpty() || !namespace.get().equals(deployServiceEntity.getNamespace())) {
-            throw new AccessDeniedException(
+            String errorMsg = String.format(
                     "No permissions to view details of services belonging to other users.");
+            log.error(errorMsg);
+            throw new AccessDeniedException(errorMsg);
         }
         return EntityTransUtils.transDeployServiceEntityToServiceDetailVo(deployServiceEntity);
     }
 
+    /**
+     * Get vendor hosted service detail by id.
+     *
+     * @param id ID of deploy service.
+     * @return VendorHostedServiceDetailsVo
+     */
+    public VendorHostedServiceDetailsVo getVendorHostedServiceDetailsById(UUID id) {
+        ServiceDetailVo serviceDetailVo = getServiceDetailsByIdForIsv(id);
```

Comment:
also since this new method is for end users only, calling this existing method will not work since the end user will not have any namespace information.